contentcuration/contentcuration/static/js/edit_channel/channel_settings/views.js

```diff
@@ -0,0 +1,142 @@
+        this.thumbnail_url = this.model.get("thumbnail_url");
+        this.thumbnail = this.model.get("thumbnail");
+        this.render();
+    },
+    events: {
+      "click #settings_submit" : "submit_changes",
+      "change .input_listener" : "register_changes",
+      "keyup .input_listener" : "register_changes"
+    },
+    render: function() {
+        this.$el.html(this.template({
+            channel: this.model.toJSON(),
+            licenses: window.licenses.toJSON(),
+            preferences: this.model.get('preferences')
+        }));
+        $("#license_select").val(this.get_license_id(this.model.get('preferences').license));
+        $("#mastery_model_select").val(this.model.get('preferences').mastery_model);
+        $("#custom_license_description").css('display', (this.get_license_name()==="Special Permissions")? "block" : "none");
+        $("#mastery_custom_criterion").css('visibility', ($("#mastery_model_select").val()==="m_of_n")? "visible" : "hidden");
+        this.image_upload = new FileViews.ThumbnailUploadView({
+            model: this.model,
+            el: this.$("#channel_thumbnail"),
+            preset_id: 'channel_thumbnail',
+            upload_url: window.Urls.thumbnail_upload(),
+            acceptedFiles: 'image/jpeg,image/jpeg,image/png',
+            image_url: this.thumbnail_url,
+            default_url: "/static/img/kolibri_placeholder.png",
+            onsuccess: this.set_thumbnail,
+            onerror: this.reset_thumbnail,
+            oncancel:this.enable_submit,
+            onstart: this.disable_submit,
+            onremove: this.remove_thumbnail,
+            allow_edit: true
+        });
+    },
+    get_license_id: function(license_name){
+        return window.licenses.findWhere({license_name: license_name}).id;
+    },
+    get_license_name: function(){
+        return window.licenses.get($("#license_select").val()).get('license_name');
+    },
+    submit_changes:function(){
+        var preferences = this.model.get('preferences');
+        preferences["license"] = this.get_license_name();
+        preferences["author"] = $("#author_field").val().trim();
+        preferences["copyright_holder"] = $("#input_copyright_holder").val().trim();
+        preferences["license_description"] = $("#custom_license_description").val().trim();
+        preferences["mastery_model"] = $("#mastery_model_select").val();
+        preferences["m_value"] = $("#m_value").val();
+        preferences["n_value"] = $("#n_value").val();
+        preferences["auto_derive_video_thumbnail"] = $("#auto_video_thumbnail").is(":checked");
+        preferences["auto_derive_audio_thumbnail"] = $("#auto_audio_thumbnail").is(":checked");
+        preferences["auto_derive_document_thumbnail"] = $("#auto_document_thumbnail").is(":checked");
+        preferences["auto_derive_html5_thumbnail"] = $("#auto_html5_thumbnail").is(":checked");
+        var self = this;
+        $("#settings_submit").html("Saving...");
```

Comment:
Are these jquery methods chainable? Could chain `attr` and `addClass` from this line.